Cron/IndexBlogUrls.php

```diff
@@ -0,0 +1,203 @@
+        $this->wordpressPostRepository = $wordpressPostRepository;
+    }
+
+    public function execute(): void
+    {
+        $sites = $this->remotePostRepository->getList();
+        $dbPosts = $this->wordpressPostRepository->getItems();
+
+        /* @var string[] $urls */
+        $urls = [];
+        /* @var WordpressPostData[] $toBeDeleted */
+        $toBeDeleted = [];
+        /* @var array $toBeAdded */
+        $toBeAdded = [];
+
+        foreach ($sites as $siteId => $site) {
+            $siteOptions = $this->siteRepository->get((string)$siteId);
+
+            if (!$siteOptions->getEnableBlog()) {
+                // * Site is disabled in M2
+                $toBeDeleted = array_merge($toBeDeleted, array_filter($dbPosts, function ($row) use ($siteId) {
+                    return (int)$row->getSiteId() === $siteId;
+                }));
+                continue;
+            }
+
+            foreach ($site['data'] as $post) {
+                $dbPost = array_filter($dbPosts, function ($row) use ($post, $siteId) {
+                    return (int)$row->getPostId() === $post['id'] &&
+                        (int)$row->getSiteId() === $siteId;
+                });
+
+                if (count($dbPost) === 0) {
+                    // * Page is added in WP
+                    $toBeAdded[] = array_merge($post, ['site_id' => $siteId]);
+                    continue;
+                }
+
+                $dbPost = $dbPost[0];
+                if ($dbPost->getSlug() !== $post['slug']) {
+                    $urls = array_merge($urls, $this->updatePageSlug($dbPost, $post['slug']));
+                }
+            }
+        }
+
+        foreach ($dbPosts as $post) {
+            // * Page has been deleted from WP
+            $site = $sites[$post->getSiteId()];
+
+            $isDeleted = array_filter($site['data'], function ($row) use ($post) {
+                return $row['id'] === (int)$post->getPostId();
+            });
+
+            if (count($isDeleted) !== 0) {
+                continue;
+            }
+
+            $toBeDeleted[] = $post;
+        }
+
+        foreach ($toBeAdded as $row) {
+            $urls = array_merge($urls, $this->registerPage($row));
+        }
+
+        foreach ($toBeDeleted as $row) {
+            $this->unregisterPage($row);
+        }
+
+        $this->urlPersistInterface->replace($urls);
+    }
+
+    public function unregisterPage(WordpressPostData $page)
+    {
+        // * To be deleted
+        $this->urlPersistInterface->deleteByData([
+            UrlRewrite::ENTITY_TYPE => self::URLREWRITE_ENTITY_TYPE,
+            UrlRewrite::ENTITY_ID => $page->getWordpressPostId()
+        ]);
+
+        $this->wordpressPostRepository->delete($page->getWordpressPostId());
+    }
+
+    public function registerPage(array $page)
+    {
+        // * To be added
+        [
+            'site_id' => $siteId,
+            'id' => $postId,
+            'slug' => $slug
+        ] = $page;
```

Comment:
I may be ignorant of this but, so enlighten me: Why is this variable declared in this specific way?

Reply:
It felt cleaner to me to deconstruct the $page variable, and just use the 3 variables separately across the function, rather than keeping to do `$page['site_id']` etc.